staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,315 @@
+		}{
+			args:    alias.Args,
+			flags:   alias.Flags,
+			command: aliasCmd,
+		}
+	}
+
+	// It is verified that all the aliases are valid. So that, now we
+	// can define them in to the root command.
+	for key, val := range aliasArgsMap {
+		p.aliases[key] = struct{}{}
+		rootCmd.AddCommand(val.command)
+	}
+
+	aliasName := args[1]
+
+	foundAliasCmd, _, err := rootCmd.Find([]string{aliasName})
+	if err != nil {
+		return args, nil
+	}
+
+	aliasArgs, ok := aliasArgsMap[aliasName]
+	if !ok {
+		return args, nil
+	}
+
+	// This function triggers merging the persistent flags in the parent commands.
+	_ = foundAliasCmd.InheritedFlags()
+
+	for _, fl := range aliasArgs.flags {
+		// explicit flag usage has higher precedence than the kuberc default flag value.
+		// We should set the default flag values in kuberc, unless there is any in args.
+		if explicitFlagUse(fl.Name, args) {
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			fmt.Fprintf(errOut, "could not apply value %s to flag %s in alias %s err: %v\n", fl.Default, fl.Name, args[0], err)
+			return args, nil
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
+	return args, nil
+}
+
+// DefaultGetPreferences returns v1alpha1.KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
+// where kubeconfig's defaults are residing in.
+func DefaultGetPreferences(kuberc string) (*v1alpha1.Preferences, error) {
+	if !util.KubeRC.IsEnabled() {
+		return nil, nil
+	}
+
+	kubeRCFile := RecommendedKubeRCFile
+	explicitly := false
+	if kuberc != "" {
+		kubeRCFile = kuberc
+		explicitly = true
+	}
+
+	if kubeRCFile == "" && os.Getenv("KUBERC") != "" {
+		kubeRCFile = os.Getenv("KUBERC")
+		explicitly = true
+	}
+
+	kubeRCBytes, err := os.ReadFile(kubeRCFile)
+	if err != nil {
+		if os.IsNotExist(err) && !explicitly {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	decoded, err := runtime.Decode(scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion), kubeRCBytes)
+	if err != nil {
+		return nil, err
+	}
+	return decoded.(*v1alpha1.Preferences), nil
+}
+
+func getExplicitKuberc(args []string) string {
+	var kubercPath string
+	for i, arg := range args {
+		if arg == "--kuberc" {
+			if i+1 < len(args) {
+				kubercPath = args[i+1]
+				break
+			}
+		} else if strings.Contains(arg, "--kuberc=") {
+			parg := strings.Split(arg, "=")
+			if len(parg) > 1 {
+				kubercPath = parg[1]
+				break
+			}
+		}
+	}
+
+	if kubercPath == "" {
+		return ""
+	}
+
+	return kubercPath
+}
+
+func explicitFlagUse(flagName string, args []string) bool {
+	for _, arg := range args {
+		if strings.HasPrefix(arg, fmt.Sprintf("--%s", flagName)) || strings.HasPrefix(arg, fmt.Sprintf("--%s=", flagName)) {
```

Comment:
I think there is something wrong here, as it will match partial flag names.

For example, I set an override in my kuberc, like this:
```
    - name: user
      default: "brian"
```

But this function matches on the `--username` flag I explicitly set (I chose this example because there is both a `--user` and a `--username` flag):
```
_output/bin/kubectl get pod --username=test
```

I had to add some printf's in the code to see what was going on, so I think maybe some logging might be helpful too, for troubleshooting.

Reply:
You always see what I'm missing and I really appreciate to have a look at this PR. It should look at the equality not `HasPrefix`.

Reply:
This code block should catch `--username=test` and it is clearly not equal to `user` flag. But there was an issue in the code that `--username test` matches to `--user` flag and it is fixed now.

Reply:
Added a test case to also catch incorrect partial flag matching.